src/familiar.ts

```diff
@@ -55,177 +63,326 @@ export function meatFamiliar(): Familiar {
+
+  const compareFams = (a: GeneralFamiliar, b: GeneralFamiliar) => {
+    if (a.expectedValue === b.expectedValue) {
+      return findLeprechaunMultiplier(a.familiar) > findLeprechaunMultiplier(b.familiar);
     }
-  }
+    return a.expectedValue > b.expectedValue;
+  };
 
-  if (have($familiar`Stocking Mimic`)) {
-    const mimicWeight = myFamiliarWeight($familiar`Stocking Mimic`);
-    const actionPercentage = 1 / 3 + (haveEffect($effect`Jingle Jangle Jingle`) ? 0.1 : 0);
-    const mimicValue = mimicDropValue() + ((mimicWeight * actionPercentage * 1) / 4) * 10 * 4 * 1.2;
-    familiarValue.push([$familiar`Stocking Mimic`, mimicValue]);
-  }
+  return familiars.reduce((a, b) => (compareFams(a, b) ? a : b));
+}
 
-  if (have($familiar`Obtuse Angel`)) {
-    familiarValue.push([$familiar`Obtuse Angel`, 0.02 * garboValue($item`time's arrow`)]);
-  }
+export function freeFightFamiliar(canMeatify = false): Familiar {
+  return freeFightFamiliarData(canMeatify).familiar;
+}
 
-  if (have($familiar`Robortender`)) familiarValue.push([$familiar`Robortender`, 200]);
+type MarginalFamiliar = GeneralFamiliar & { outfitValue: number };
+const outfitSlots = $slots`hat, back, shirt, weapon, off-hand, pants, acc1, acc2, acc3, familiar`;
+const cachedOutfits = new Map<number, { weight: number; meat: number; bonus: number }>();
 
-  for (const familiar of $familiars`Hobo Monkey, Cat Burglar, Urchin Urchin, Leprechaun`) {
-    if (have(familiar)) familiarValue.push([familiar, 1]);
-  }
+function cacheOutfit(
+  mult: number,
+  familiar: Familiar
+): { weight: number; meat: number; bonus: number } {
+  useFamiliar(familiar);
+  meatOutfit(false, undefined, false);
+  const outfit = outfitSlots.map((slot) => equippedItem(slot));
+  const bonuses = bonusGear("barf");
+  const values = {
+    weight: sum(outfit, (eq: Item) => getModifier("Familiar Weight", eq)),
+    meat: sum(outfit, (eq: Item) => getModifier("Meat Drop", eq)),
+    bonus: sum(outfit, (eq: Item) => bonuses.get(eq) ?? 0),
+  };
+  cachedOutfits.set(mult, values);
+  return values;
+}
+
+export function setMarginalFamiliar(): void {
+  const buffWeight = sum(getActiveEffects(), (ef: Effect) => getModifier("Familiar Weight", ef));
+  const outfitWeight = sum(outfitSlots, (slot: Slot) =>
+    getModifier("Familiar Weight", equippedItem(slot))
+  );
+  const passiveWeight = weightAdjustment() - buffWeight - outfitWeight;
+
+  const dropFamiliars: MarginalFamiliar[] = [
+    ...filterNull(rotatingFamiliars.map(valueStandardDropFamiliar)),
+    ...standardFamiliars(),
+    {
+      familiar: $familiar`Space Jellyfish`,
+      expectedValue:
+        garboValue($item`stench jelly`) /
+        (get("_spaceJellyfishDrops") < 5 ? get("_spaceJellyfishDrops") + 1 : 20),
+      leprechaunMultiplier: 0,
+    },
+  ].map(({ familiar, expectedValue, leprechaunMultiplier }) => {
+    const { meat, weight, bonus } =
+      cachedOutfits.get(leprechaunMultiplier) ?? cacheOutfit(leprechaunMultiplier, familiar);
```

Comment:
According to Dict, fairychauns are worth 7/6th of a normal lep (or rather, fairies are 1/6th of a lep). I didn't really have time today to work through stuff proper, but I don't know if you intend to group that into the `leprechaunMultiplier`, or if you eventually want to end up with a structure that handles the "hashing of a tuple"